src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,147 @@
+
+import java.util.Scanner;
+
+import static duke.exceptions.UserInputException.inputExceptionType;
+import static duke.exceptions.UserInputException.inputExceptionType.INVALID_TASK_TYPE;
+
+public class Duke {
+    //String constants
+    private static final String LOGO = " ____        _        \n" + "|  _ \\ _   _| | _____ \n" + "| | | | | | | |/ / _ \\\n" + "| |_| | |_| |   <  __/\n" + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String HELLO_MESSAGE = " Hello! I'm Duke\n" + "     What can I do for you?\n";
+    private static final String BYE_MESSAGE = " Bye. Hope to see you again soon!";
+    private static final String SEPARATOR = "____________________________________________________________";
+    private static final String COMMAND_LIST_WORD = "list";
+    private static final String COMMAND_MARK_WORD = "mark";
+    private static final String COMMAND_UNMARK_WORD = "unmark";
+    private static final String COMMAND_ADD_TODO_WORD = "todo";
+    private static final String COMMAND_ADD_DEADLINE_WORD = "deadline";
+    private static final String COMMAND_ADD_EVENT_WORD = "event";
+    private static final String COMMAND_EXIT_WORD = "bye";
+    private static final String LINE_PREFIX = "    ";
+    //Scanner for user input
+    private static final Scanner SCANNER = new Scanner(System.in);
+    //Task list
+    private static final Task[] TASK_LIST = new Task[100];
+    private static int taskCount = 0;
+
+    public static void main(String[] args) throws DukeException {
+        // reused from contacts Contacts1.java with modification
+        showWelcomeMessage();
+        while (true) {
+            try{
+                String userCommand = getUserInput();
+                executeCommand(userCommand);
+            }catch(DukeException err){
+                showToUser(err.ProduceErrorMessage());
+                showToUser(SEPARATOR);
+            }
+        }
+    }
+
+    private static void showWelcomeMessage() {
+        showToUser("Hello from\n" + LOGO);
+        showToUser(SEPARATOR);
+        showToUser(HELLO_MESSAGE);
+        showToUser(SEPARATOR);
+    }
+
+    private static String getUserInput() {
+        return SCANNER.nextLine();
+    }
+
+    private static void executeCommand(String userInputString) throws DukeException {
+        showToUser(SEPARATOR);
+        final String[] commandTypeAndParams = splitCommandWordAndArgs(userInputString);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+        switch (commandType) {
+        case COMMAND_LIST_WORD:
+            listTask();
+            break;
+        case COMMAND_MARK_WORD:
+            markAsDone(commandArgs);
+            break;
+        case COMMAND_UNMARK_WORD:
+            markAsNotDone(commandArgs);
+            break;
+        case COMMAND_EXIT_WORD:
+            exitProgram();
+            break;
+        case COMMAND_ADD_TODO_WORD:
+        case COMMAND_ADD_DEADLINE_WORD:
```

Comment:
Missing fallthrough comments here. Take a look at coding standards doc to see what's expected in the case of fallthrough.